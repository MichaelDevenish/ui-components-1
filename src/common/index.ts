export { Props } from './types'
export { getColor } from './legacy'
