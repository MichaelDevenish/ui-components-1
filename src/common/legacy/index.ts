export { getColor } from './colors/getColor'
