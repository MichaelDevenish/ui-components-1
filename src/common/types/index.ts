/**
 * For Examples, see:
 * https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts
 * https://github.com/alm-tools/alm/blob/master/src/common/types.ts
 */
import { Size as _s, AllSizes } from './size'

namespace Props {
  export import Size = _s
}

export {
  Props,
  AllSizes
}
