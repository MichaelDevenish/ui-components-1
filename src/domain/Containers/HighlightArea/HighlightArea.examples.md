
A HighlightArea is used to highlight areas.

```jsx
import { Variables } from '@Common';

<HighlightArea color="grey">
    Hello
</HighlightArea>
```
