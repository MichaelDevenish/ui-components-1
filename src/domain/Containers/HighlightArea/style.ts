import styled from 'styled-components'

import { Variables } from '../../../common'
import { styleForMargins } from '../../Spacers/services/margins'
import { HighLightAreaColors, IHighlightAreaProps } from './HighlightArea'

interface IColorOption {
  border: Variables.Color,
  background: Variables.Color,
}

type ColorOptions = {
  [K in HighLightAreaColors]: IColorOption
}

const colorOptions: ColorOptions = {
  grey: {
    border: Variables.Color.n400,
    background: Variables.Color.n150
  },
  blue: {
    border: Variables.Color.b200,
    background: Variables.Color.b100
  }
}

const StyledHighlightArea = styled.div<IHighlightAreaProps>`
  border: 1px solid ${(props: IHighlightAreaProps) => colorOptions[props.color].border};
  background-color: ${(props: IHighlightAreaProps) => colorOptions[props.color].background};
  padding: ${Variables.Spacing.sMedium}px;
  border-radius: ${Variables.Style.borderRadius}px;
  ${(props: IHighlightAreaProps) => styleForMargins(props.margins)}
`

export {
  StyledHighlightArea
}
