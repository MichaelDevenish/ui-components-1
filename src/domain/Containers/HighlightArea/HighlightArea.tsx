import React from 'react'

import { Props } from '../../../common'
import { StyledHighlightArea } from './style'

type Color = 'grey' | 'gray'

interface IHighlightAreaProps {
  /** Background color of the highlight area */
  color: Color
  /** Margins around the highlight area */
  margins?: Props.IMargins
  /** Children components */
  children: React.ReactNode
  /** The component context */
  componentContext?: string
}

const HighlightArea: React.FC<IHighlightAreaProps> = ({ componentContext, color, margins, children }) => {
  return (
    <div
      data-component-type={Props.ComponentType.HightlightArea}
      data-component-context={componentContext}
    >
      <StyledHighlightArea color={color} margins={margins}>
        {children}
      </StyledHighlightArea>
    </div>
  )
}

export {
  Color,
  IHighlightAreaProps,
  HighlightArea
}
