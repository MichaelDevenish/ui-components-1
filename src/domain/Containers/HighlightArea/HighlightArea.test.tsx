import { mount } from 'enzyme'
import React from 'react'

import { Variables } from '../../../common'
import { HighlightArea } from './HighlightArea'

describe('<HighLightArea />', () => {
    it(`should render a HighlightArea`, () => {
        const wrapper = mount(
            <HighlightArea
                color='grey'
                margins={Variables.Spacing.sMedium}
            >
                Hello
            </HighlightArea>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
