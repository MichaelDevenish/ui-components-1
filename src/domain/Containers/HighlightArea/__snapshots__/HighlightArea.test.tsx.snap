// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighLightArea /> should render a HighlightArea 1`] = `
.c0 {
  border: 1px solid #C5CED6;
  background-color: #FAFBFC;
  padding: 16px;
  border-radius: 4px;
}

<HighlightArea
  color="grey"
>
  <styled.div
    color="grey"
    data-component-type="highlight_area"
  >
    <StyledComponent
      color="grey"
      data-component-type="highlight_area"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  border: 1px solid ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  padding: ",
              "16",
              "px;
  border-radius: ",
              "4",
              "px;
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
        color="grey"
        data-component-type="highlight_area"
      >
        Hello
      </div>
    </StyledComponent>
  </styled.div>
</HighlightArea>
`;
